### PR TITLE
Set cookies preferences

### PR DESCRIPTION
**Pull request by cpjmcquillan**
This PR allows users to set and update their cookie preferences

## Jira ticket URL
https://dfedigital.atlassian.net/browse/TEVA-1232

## Changes in this PR
- User can set cookies preference
- Cookies banner is only rendered if cookies preference has not been set
- Only run GTM script if user has consented to cookies

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -17,15 +17,94 @@ en:
...
+          like your IP address or your postcode before we share it with Google.
+        details:
+          intro: 'Google Analytics stores information about:'
+          list:
+            - the pages you visit on Teaching Vacancies
+            - how long you spend on each page
+            - how you got to the site
+            - what you click on while you're visiting the site
+        explanation: >-
+          We use Google Analytics software to collection information about how you use Teaching Vacancies.
```
Comment (stevenleggdfe) [inline config/locales/en.yml:52]: @cpjmcquillan typo here - 'collection' should be collect. To check, is this the final version of the content from @valentine-carter ?

File: app/views/layouts/application.html.haml [hunk 2]
```diff
@@ -29,11 +29,11 @@
     %meta{ content: asset_pack_path('media/images/govuk-opengraph-image.png'), property: "og:image" }/
     = csrf_meta_tags
   %body{ class: body_class, data: { vacancy_state: @vacancy.present? ? @vacancy.state : 'create' } }
-    = render partial: 'shared/google_tag_manager_body'
+    = render partial: 'shared/google_tag_manager_body' if !CookiesBannerFeature.enabled? || cookies['consented-to-cookies'] == 'yes'
     :javascript
       document.body.className = ((document.body.className) ? document.body.className + ' js-enabled' : 'js-enabled');
     = render 'shared/skip_links'
-    = render 'layouts/cookies_banner' if CookiesBannerFeature.enabled? && !current_page?(page_path('cookies'))
+    = render 'layouts/cookies_banner' unless !CookiesBannerFeature.enabled? || controller_name == 'cookies_preferences' || cookies_preference_set?
```
Comment (david-mears-dfe) [inline app/views/layouts/application.html.haml:36]: I don't know how much time it takes for an app to check a little variable for true/false (probably almost none), but we might shave off a millisecond or two by putting the most-frequently-true thing to evaluate first (`cookies_preference_set?`).
🤷

File: config/locales/activerecord.yml [hunk 3]
```diff
@@ -65,6 +65,9 @@ en:
   hiring_staff_user_preference_errors:
     managed_organisations:
       blank: Select which schools' job listings you would like to view
+  cookies_preferences_errors: &cookies_preferences_errors
```
Comment (david-mears-dfe) [inline config/locales/activerecord.yml:68]: The ampersand denotes an alias, yes? I don't see a difference here...?

**Review by david-mears-dfe — APPROVED** [review]

File: app/views/layouts/application.html.haml (hunk 2 above)
Comment (cpjmcquillan) [reply app/views/layouts/application.html.haml:36]: The order is important. 

If the feature is disabled, **don't render**
If the feature is enabled, but we're at the cookies_preferences create/new action, **don't render**
If the feature is enabled, and we're not at the cookies_preferences create/new action, but cookies preferences are set, **don't render**
If the feature is enabled, and we're not at the cookies_preferences create/new action, and cookies preferences are not set, **render**

**Review by david-mears-dfe — CHANGES_REQUESTED** [review]
Just thought, I'm not sure if you actually initialize CookiesBannerFeature?

**Comment by cpjmcquillan** [comment]
> Just thought, I'm not sure if you actually initialize CookiesBannerFeature?

Been and done and merged to master

**Review by david-mears-dfe — APPROVED** [review]

**Pull request merged**